packages/common/src/api/library.ts

```diff
@@ -0,0 +1,74 @@
+import type { full } from '@audius/sdk'
+
+import { createApi } from 'audius-query'
+import { UserCollectionMetadata } from 'models/Collection'
+import { Kind } from 'models/Kind'
+import { makeActivity } from 'services/audius-api-client/ResponseAdapter'
+import { APIActivityV2 } from 'services/audius-api-client/types'
+import { reformatCollection } from 'store/cache/collections/utils/reformatCollection'
+import { encodeHashId } from 'utils/hashIds'
+import { removeNullable } from 'utils/typeUtils'
+
+type GetLibraryAlbumsArgs = {
+  userId: number
+  offset: number
+  limit: number
+  query?: string
+  sortMethod?: full.GetUserLibraryAlbumsSortMethodEnum
+  sortDirection?: full.GetUserLibraryAlbumsSortDirectionEnum
+}
+
+export const libraryApi = createApi({
+  reducerPath: 'libraryApi',
+  endpoints: {
+    getLibraryAlbums: {
+      fetch: async (
+        args: GetLibraryAlbumsArgs,
+        { audiusSdk, audiusBackend }
+      ) => {
+        const {
+          userId,
+          offset,
+          limit,
+          query = '',
+          sortMethod = 'added_date',
+          sortDirection = 'desc'
+        } = args
+        const sdk = await audiusSdk()
+        const { data, signature } =
+          await audiusBackend.signDiscoveryNodeRequest()
```

Comment:
Do we have to do this for every sdk request? Feels like it should be baked into the sdk :-/

Reply:
yeah 100%, it's a follow up task for the SDK https://linear.app/audius/issue/C-2928/enable-sdk-reads-for-authenticated-routes-[follow-on]